server/controllers/RatingController.js

```diff
@@ -0,0 +1,176 @@
+   * @param { Function } next error handling middleware
+   * @returns {Object} res.body contain the info of the rating
+   * res.body is object with properties ratingDetail and result
+   */
+  static rateArticle(req, res, next) {
+    let { rating } = req.body;
+    let userId = req.body.decoded.id;
+    const { slug } = req.params;
+    rating = parseInt(rating, 10);
+    userId = parseInt(userId, 10);
+    return Article
+      .find({
+        where: {
+          slug,
+        },
+        include: [{ model: Rating, as: 'ratings' }],
+      })
+      .then((article) => {
+        if (article) {
+          if (article.authorId === userId) {
+            return res.status(403).json({
+              status: 'fail',
+              message: 'You can\'t provide ratings for your article',
+            });
+          }
+          return Rating
+            .findOrCreate({
+              where: {
+                articleId: article.id,
+                userId,
+              },
+              defaults: { rating },
+            })
+            .spread((rateData, rated) => {
+              if (rated) {
+                article.ratings.push(rateData);
+                // call function to compute the average of rating
+                const ratingInfo = ratingServices
+                  .computeRating(article.ratings);
+
+                // add property to ratingInfo object
+                ratingInfo.ratingDetail.articleId = article.id;
+                ratingInfo.ratingDetail.authorId = article.authorId;
+                ratingInfo.rating = rateData;
+                return res.status(201).json(ratingInfo);
+              }
+              return res.status(403).json({
+                status: 'fail',
+                message: 'Your rating have already been recoded',
+              });
+            });
+        }
+        return res.status(404).json({
+          status: 'fail',
+          message: 'Article not found',
+        });
+      })
+      .catch(next);
+  }
+
+  /**
+   * post the rating of the article
+   * with slug given in req.params
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param { function } next call error handling middleware
+   * @returns {Object} res.body
+   * res.body is object with properties ratingDetail and result
+   */
+  static updateRating(req, res, next) {
+    let { rating } = req.body;
+    let userId = req.body.decoded.id;
+    const { slug } = req.params;
+    rating = parseInt(rating, 10);
+    userId = parseInt(userId, 10);
+    return Article
+      .find({
+        where: {
+          slug,
+        },
+        include: [{ model: Rating, as: 'ratings' }],
+      })
+      .then((article) => {
+        if (article) {
+          if (article.authorId === userId) {
+            return res.status(403).json({
+              status: 'fail',
+              message: 'You can\'t provide ratings for your article',
+            });
+          }
```

Comment:
I have noticed that this logic is repeated twice. I think you should abstract part of it to a new helper function so that you would increase `maintanability`